src/integration-test/java/uk/gov/hmcts/juror/api/moj/report/bespoke/YieldPerformanceReportITest.java

```diff
@@ -0,0 +1,330 @@
+    "/db/mod/truncate.sql",
+    "/db/mod/reports/YieldPerformanceReportITest_typical.sql",
+})
+@SuppressWarnings("PMD.JUnitTestsShouldIncludeAssert"//False positive
+)
+class YieldPerformanceReportITest extends AbstractControllerIntegrationTest<YieldPerformanceReportRequest,
+    YieldPerformanceReportResponse> {
+    public static final String URL = "/api/v1/moj/reports/yield-performance";
+    public static final String LOCAL_DATE = "LocalDate";
+
+    @Autowired
+    public YieldPerformanceReportITest(TestRestTemplate template) {
+        super(HttpMethod.POST, template, HttpStatus.OK);
+    }
+
+    @Override
+    protected String getValidUrl() {
+        return URL;
+    }
+
+    @Override
+    protected String getValidJwt() {
+        return createJwt(
+            "test_bureau_standard",
+            "400",
+            UserType.BUREAU,
+            Set.of(),
+            "400"
+        );
+    }
+
+    @Override
+    protected YieldPerformanceReportRequest getValidPayload() {
+        return null;
+    }
+
+    @Test
+    void viewByCourt() {
+
+        YieldPerformanceReportRequest payload =  YieldPerformanceReportRequest.builder()
+            .allCourts(false)
+            .courtLocCodes(List.of("415"))
+            .fromDate(LocalDate.parse("2024-08-01"))
+            .toDate(LocalDate.parse("2024-08-20"))
+            .build();
+
+        testBuilder()
+            .payload(payload)
+            .triggerValid()
+            .responseConsumer(this::verifyHeadingsCourt)
+            .responseConsumer(this::verifyCourtPayload);
+
+    }
+
+    @Test
+    void viewByCourts() {
+
+        YieldPerformanceReportRequest payload =  YieldPerformanceReportRequest.builder()
+            .allCourts(false)
+            .courtLocCodes(List.of("415","774"))
+            .fromDate(LocalDate.parse("2024-08-01"))
+            .toDate(LocalDate.parse("2024-08-20"))
+            .build();
+
+        testBuilder()
+            .payload(payload)
+            .triggerValid()
+            .responseConsumer(this::verifyHeadingsCourt)
+            .responseConsumer(this::verifyCourtsPayload);
+
+    }
+
+    @Test
+    void viewByAllCourts() {
+
+        YieldPerformanceReportRequest payload =  YieldPerformanceReportRequest.builder()
+            .allCourts(true)
+            .fromDate(LocalDate.parse("2024-07-21"))
+            .toDate(LocalDate.parse("2024-08-07"))
+            .build();
+
+        testBuilder()
+            .payload(payload)
+            .triggerValid()
+            .responseConsumer(this::verifyHeadingsAllCourts)
+            .responseConsumer(this::verifyAllCourtsPayload);
+
+    }
+
+    @Test
+    void unhappyCourtUser() {
+
+        YieldPerformanceReportRequest payload =  YieldPerformanceReportRequest.builder()
+            .allCourts(false)
+            .courtLocCodes(List.of("415"))
+            .fromDate(LocalDate.parse("2024-08-01"))
+            .toDate(LocalDate.parse("2024-08-20"))
+            .build();
```

Comment:
Not needed should default to value from getValidPayload()

Reply:
each test needs a different payload so using this override

Reply:
The negative test should be able to use the standard getValidPayload method instead of creating a bespoke one as the test only needs a valid payload

Reply:
agreed, updated.